buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -715,10 +715,15 @@ class ClusterFormationTasks {
         wait.doLast {
 
             Collection<String> unicastHosts = new HashSet<>()
-            nodes.forEach { otherNode ->
-                String unicastHost = otherNode.config.unicastTransportUri(otherNode, null, project.ant)
-                if (unicastHost != null) {
-                    unicastHosts.addAll(Arrays.asList(unicastHost.split(",")))
+            nodes.forEach { node ->
+                List<String> manualSeedUris = node.config.manualSeedUris.call()
+                if (manualSeedUris.empty == false) {
+                    unicastHosts.addAll(manualSeedUris)
+                } else {
```

Comment:
I think the unicast hosts file should be the _union_ of these, so no need for the `else`.

Reply:
I think I agree, unfortunately this forces us to manually set min master nodes (just tried it out). The reason being that the nodes that we have join the cluster effectively think they're in a 1 node cluster => if we add the node itself to the unicast hosts here it blows up => I'll add the min master nodes setting back but set it to  `2` maybe so it's more realistic + make it merge the lists here?

Reply:
We can still drop the min master node setting for the initial `oldCluster` though :)

Reply:
Bah of course, it's a different cluster from Gradle's point of view, so yes we need to set `minimum_master_nodes` on the new nodes.

I'm surprised that adding the node to its own unicast hosts list makes a difference, I think this would be broken either way:

https://github.com/elastic/elasticsearch/blob/93c2c604e57cae4ba093c7053dec604a33d89e36/server/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java#L213-L214

Reply:
Hmm it did work when I ran it locally :D but yea, set them to `2` now on the 1 node "clusters" and it passed fine locally.